python/lsst/validate/drp/report_performance.py

```diff
@@ -0,0 +1,158 @@
+    srd_level : SRD level to quote.  One of ['design', 'minimum', 'stretch']
+    release_metrics_file : filepath of YAML file of a given release level.
+       The JSON file will store the metrics it was calculated with
+       But one will wish to compare against an external set of specifications.
+       Such as the release for a given version during a given fiscal year.
+    release_level : E.g., 'FY17' or 'ORR'
+
+    Products
+    ---
+    Writes table of performance metrics to an RST file.
+    """
+    input_objects = ingest_data(validation_drp_report_filenames)
+    input_table = objects_to_table(input_objects, level=srd_level)
+    if release_metrics_file is not None and release_level is not None:
+        release_metrics = load_metrics(release_metrics_file)
+        add_release_metric(input_table, release_metrics, release_level)
+
+    write_report(input_table, output_file)
+
+
+def ingest_data(filenames):
+    """"""
+    jobs = {}
+    # Read in JSON output from metrics run
+    for file in filenames:
+        job = load_json_output(file)
+        filter_name = get_filter_name_from_job(job)
+        jobs[filter_name] = job
+
+    return jobs
+
+
+# Identify key data from JSON
+def objects_to_table(input_objects, level='design'):
+    """Take objects and convert to table."""
+    rows = []
+    for filter_name, obj in input_objects.items():
+        for meas in obj.measurements:
+            # Skip specification levels (called .name in measurement objects)
+            #  that are not the 'level' we are looking for.
+            if meas.spec_name is not None and meas.spec_name != level:
+                continue
+            m = meas.metric
+            spec = m.get_spec(level, filter_name=filter_name)
+            if meas.quantity is None:
+                meas_quantity_value = "--"
+            else:
+                meas_quantity_value = meas.quantity.value
+            this_row = [m.name, filter_name, meas_quantity_value, meas.unit,
+                        m.operator_str, spec.quantity.value]
+            rows.append(this_row)
+
+    srd_requirement_col_name = 'SRD Requirement: %s' % level
+    col_names = ('Metric', 'Filter', 'Value', 'Unit',
+                 'Operator', srd_requirement_col_name)
+    output = Table(rows=rows, names=col_names)
+    output.add_column(Column(['']*len(output), dtype=str, name='Comments'))
+    return output
+
+
+# Calculate numbers in table
+def add_release_metric(data, release_metrics, release_metrics_level):
+    release_targets = []
+    for row in data:
+        metric = release_metrics[row['Metric']]
+        spec = metric.get_spec(
+            name=release_metrics_level, filter_name=row['Filter'])
+        release_targets.append(spec.quantity.value)
+
+    release_targets_col = Column(
+        release_targets,
+        dtype=float,
+        name='Release Target: %s' % release_metrics_level)
+    data.add_column(release_targets_col)
+
+    return data
```

Comment:
I think it's confusing to both modify in place and return the object.  This could lead to patterns like:
```
data2 = add_release_metric(data1, metrics, level)
```
The above implies that data1 is unchanged.  I'd prefer if this method simply didn't have a `return` statement.